src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/Rechargeable.java

```diff
@@ -0,0 +1,145 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import io.github.thebusybiscuit.slimefun4.core.networks.energy.EnergyNet;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.gadgets.Jetpack;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.gadgets.MultiTool;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.ChargingBench;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+
+/**
+ * A {@link Rechargeable} {@link SlimefunItem} can hold energy and is able to
+ * be recharged using a {@link ChargingBench}.
+ * Any {@link SlimefunItem} which is supposed to be chargeable <b>must</b> implement this interface.
+ * 
+ * @author TheBusyBiscuit
+ * 
+ * @see ChargingBench
+ * @see EnergyNet
+ * @see Jetpack
+ * @see MultiTool
+ *
+ */
+@FunctionalInterface
+public interface Rechargeable extends ItemAttribute {
+
+    /**
+     * This method returns the maximum charge the given {@link ItemStack} is capable of holding.
+     * 
+     * @param item
+     *            The {@link ItemStack} for which to determine the maximum charge
+     * 
+     * @return The maximum energy charge for this {@link ItemStack}
+     */
+    float getMaxItemCharge(ItemStack item);
```

Comment:
Should we just make these SlimefunItemStacks? I don't see a world where this would ever be on a normal item and that way we're also enforcing ID (so performance improvements) to be there.

Reply:
This will almost never be a SlimefunItemStack, unless it's factory new.